hr_expense_petty_cash/models/hr_expense.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class HrExpense(models.Model):
+    _inherit = 'hr.expense'
+
+    payment_mode = fields.Selection(
+        selection_add=[('petty_cash', 'Petty Cash')],
+    )
+    petty_cash_id = fields.Many2one(
+        string='Petty cash holder',
+        comodel_name='petty.cash',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
```

Comment:
add `ondelete='restrict',`
Note: normally, please add this when you create any  many2one